src/index.test.tsx

```diff
@@ -0,0 +1,238 @@
+          <Column header="Color" accessor="color">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should provide index context', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column>
+            <IndexCell />
+          </Column>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should use custom adapter', () => {
+    const adapter = {
+      table: (props: PropsWithChildren<{}>) => {
+        return (
+          <table style={{ backgroundColor: '#ccc' }}>{props.children}</table>
+        );
+      },
+    };
+
+    const c = (
+      <TableUIAdapterContextProvider value={adapter}>
+        <DataTable data={data}>
+          <Columns>
+            <Column header="Name" accessor="name">
+              <StringCell />
+            </Column>
+          </Columns>
+          <Table>
+            <TableHeader>
+              <HeaderRow />
+            </TableHeader>
+            <TableBody>
+              <Rows>
+                <Row />
+              </Rows>
+            </TableBody>
+          </Table>
+        </DataTable>
+      </TableUIAdapterContextProvider>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should use custom key accessor', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column header="Name" accessor="name">
+            <StringCell />
+          </Column>
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows keyAccessor="id">
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should render empty column', () => {
+    const c = (
+      <DataTable data={data}>
+        <Columns>
+          <Column />
+        </Columns>
+        <Table>
+          <TableHeader>
+            <HeaderRow />
+          </TableHeader>
+          <TableBody>
+            <Rows>
+              <Row />
+            </Rows>
+          </TableBody>
+        </Table>
+      </DataTable>
+    );
+    const { unmount } = render(c);
+    unmount();
+  });
+  it('should handle single row', () => {
```

Comment:
```suggestion
  it('should render a custom Row with external data', () => {
```
And maybe pass an object to the `<Row />`